test/spec/spectest.cpp

```diff
@@ -46,34 +36,37 @@ using namespace WasmEdge;
 
 // Preprocessing for set up aliasing.
 void resolveRegister(std::map<std::string, std::string> &Alias,
-                     rapidjson::Value &CmdArray,
-                     rapidjson::Document::AllocatorType &Allocator) {
-  rapidjson::Value::ValueIterator ItMod = CmdArray.Begin();
-  for (rapidjson::Value::ValueIterator It = CmdArray.Begin();
-       It != CmdArray.End(); ++It) {
-    const auto CmdType = It->GetObject()["type"].Get<std::string>();
+                     simdjson::dom::array CmdArray) {
+  std::string_view OrgName;
+  uint64_t LastModLine;
+  for (simdjson::dom::object Cmd : CmdArray) {
+    std::string_view CmdType = Cmd["type"];
+    bool Replaced = false;
     if (CmdType == "module"sv) {
-      // Record last module in order.
-      ItMod = It;
+      // Record last module in order
+      if (Cmd["name"].get(OrgName)) {
+        OrgName = {};
+      }
+      LastModLine = Cmd["line"];
     } else if (CmdType == "register"sv) {
-      const auto NewNameStr = It->GetObject()["as"].Get<std::string>();
-      auto OrgName = ItMod->FindMember("name");
-      if (It->GetObject().HasMember("name")) {
+      std::string_view NewNameStr = Cmd["as"];
+      std::string_view Value;
+      if (!Cmd["name"].get(Value)) {
         // Register command records the original name. Set aliasing.
-        Alias.emplace(It->GetObject()["name"].Get<std::string>(), NewNameStr);
-      } else if (OrgName != ItMod->MemberEnd()) {
-        // Register command not records the original name. Get name from the
+        Alias.emplace(std::string(Value), std::string(NewNameStr));
+      } else if (!OrgName.empty()) {
+        // Register command does not record the original name. Get name from the
         // module.
-        Alias.emplace(OrgName->value.Get<std::string>(), NewNameStr);
+        Replaced = true;
+        Alias.emplace(std::string(OrgName), std::string(NewNameStr));
       }
-      if (OrgName != ItMod->MemberEnd()) {
+      if (!OrgName.empty() && !Replaced) {
         // Module has origin name. Replace to aliased one.
-        OrgName->value.SetString(NewNameStr, Allocator);
+        Alias.emplace(std::string(OrgName), NewNameStr);
```

Comment:
This behavior is not the same as the original one. The old one is to update the `OrgName`. However, your new code is replaced the `OrgName`, which is in the `Alias`.

Reply:
As I mentioned above, SIMDJSON doesn't allow us to modify JSON objects. So, I used the Alias to store the `OrgName`s and their corresponding `NewNameStr`s, which is later used [here](https://github.com/vrnimje/WasmEdge_fork/blob/172c416c068e120b02b3ed15390070a920337277/test/spec/spectest.cpp#L577-L603). I just used the Alias, which was already used to store aliases, to also store the names that needed to be modified or added.